### PR TITLE
Debug the ESLint config file

### LINKED ISSUE #403: Fix styling rules
# Description

Currently the styling rules are not applying as they should. At the moment we are using the ESLint Stylistic plugin to handle our stylistic rules, but it doesn't fire or fix any errors. For example it doesn't fix or show an error when double quotes are used.

# Requirements

- [x] Debug the ESLint Stylistic plugin rules.
- [x] Research splitting Linting and Styling 

# Dependencies

- None

### PR DESCRIPTION
**Pull request by Anissa3005**
Resolves #403 

## 🔧 What changed

Previously, the eslint config file was not applying any of the rules to our codebase. This resulted in wring linting and styling on our codebase. 

1. Rewrite the config file following the https://typescript-eslint.io/packages/typescript-eslint documentation (thanks to @theyokohamalife )

## To Do

It seems that at the moment the Vue Eslint is not yet supported for flat config files, https://github.com/vuejs/eslint-plugin-vue/issues/1291, not sure if I can get it to work.

### CODE REVIEW

**Comment by NabbeunNabi** [comment]
LGTM as a junior new on this codebase. Looks like you had your work cut out for you! 🙌

**Review by theyokohamalife — APPROVED** [review]
LGTM ✨

File: eslint.config.js [hunk 1]
```diff
@@ -1,56 +1,42 @@
...
         ignores: [
             'dist/*',
-            'output/*',
+            '.output/*',
             '.nuxt/*',
             'coverage/*',
             'cypress/videos/*',
-            '.yarn/*'
+            '.yarn/*',
+            'typedefs/gqlTypes.ts'
```
Comment (ermish) [inline eslint.config.js:15]: we don't want this to be gitignored

Comment (ermish) [reply eslint.config.js:15]: we want to know when changes are made to the api and this is one of the easiest ways to see. It also makes the build faster when it already exists. (neglible though)

File: eslint.config.js [hunk 2]
```diff
@@ -1,56 +1,42 @@
...
         ],
     },
-    // Typescript and JS Linter combined (for all the main code files)
+    eslint.configs.recommended,
+    ...tseslint.configs.recommended,
     {
-        languageOptions: {
-            parser: vuePlugin.parser,
```
Comment (ermish) [inline eslint.config.js:25]: I think this just removed the whole linting for vue files. We'll likely need to readd this back in

**Review by ermish — COMMENTED** [review]
@Anissa3005 @theyokohamalife I added some comments. We should make a new issue and fix some of these.

**Pull request merged**